pkg/roll/run_update.go

```diff
@@ -360,10 +363,18 @@ func (u *update) countHealthy(id rcf.ID, checks map[string]health.Result) (rcNod
 			// don't check health if the update isn't even done there yet
 			continue
 		}
-		if hres, ok := checks[node]; ok && hres.Status == health.Passing {
-			ret.Healthy++
+		if hres, ok := checks[node]; ok {
+			if hres.Status == health.Passing {
+				ret.Healthy++
+			} else if hres.Status != health.Unknown {
+				ret.Unhealthy++
+			}
 		}
 	}
+
+	// We set unknown here to deal with the case of nodes not being in CurrentPods()
+	ret.Unknown = ret.Desired - ret.Healthy - ret.Unhealthy
```

Comment:
More that I'd like to know why we have unknown unknowns. We should be explicit if we're guarding against something. Improving the comment to state exactly what cases we are trying to defend against would be a good first step.